impeller/entity/geometry/stroke_path_geometry.cc

```diff
@@ -38,225 +82,312 @@ Join StrokePathGeometry::GetStrokeJoin() const {
-        for (const auto& point : arc_points) {
-          vtx.position = position + point;
-          vtx_builder.AppendVertex(vtx);
-          vtx.position = position + (-point).Reflect(forward_normal);
-          vtx_builder.AppendVertex(vtx);
-        }
-      };
-      break;
+      return &StrokePathGeometry::CreateRoundCap;
     case Cap::kSquare:
-      cap_proc = [](VertexBufferBuilder<VS::PerVertexData>& vtx_builder,
-                    const Point& position, const Point& offset, Scalar scale,
-                    bool reverse) {
-        Point orientation = offset * (reverse ? -1 : 1);
-
-        VS::PerVertexData vtx;
-
-        Point forward(offset.y, -offset.x);
-
-        vtx.position = position + orientation;
-        vtx_builder.AppendVertex(vtx);
-        vtx.position = position - orientation;
-        vtx_builder.AppendVertex(vtx);
-        vtx.position = position + orientation + forward;
-        vtx_builder.AppendVertex(vtx);
-        vtx.position = position - orientation + forward;
-        vtx_builder.AppendVertex(vtx);
-      };
-      break;
+      return &StrokePathGeometry::CreateSquareCap;
   }
-  return cap_proc;
 }
 
-// static
-VertexBufferBuilder<SolidFillVertexShader::PerVertexData>
-StrokePathGeometry::CreateSolidStrokeVertices(
-    const Path& path,
-    Scalar stroke_width,
-    Scalar scaled_miter_limit,
-    const StrokePathGeometry::JoinProc& join_proc,
-    const StrokePathGeometry::CapProc& cap_proc,
-    Scalar scale) {
-  VertexBufferBuilder<VS::PerVertexData> vtx_builder;
-  auto point_buffer = std::make_unique<std::vector<Point>>();
-  // 512 is an arbitrary choice that should be big enough for most paths without
-  // needing to reallocate. If we have motivating benchmarks we should raise or
-  // lower this number, cause dnfield just made it up!
-  point_buffer->reserve(512);
-  auto polyline = path.CreatePolyline(scale, std::move(point_buffer));
+struct StrokeGenerator {
+  StrokeGenerator(const Path::Polyline& p_polyline,
+                  const Scalar p_stroke_width,
+                  const Scalar p_scaled_miter_limit,
+                  const StrokePathGeometry::JoinProc& p_join_proc,
+                  const StrokePathGeometry::CapProc& p_cap_proc,
+                  const Scalar p_scale,
+                  VertexWriter& p_vtx_builder)
+      : polyline(p_polyline),
+        stroke_width(p_stroke_width),
+        scaled_miter_limit(p_scaled_miter_limit),
+        join_proc(p_join_proc),
+        cap_proc(p_cap_proc),
+        scale(p_scale),
+        vtx_builder(p_vtx_builder) {}
+
+  void Generate() {
+    for (size_t contour_i = 0; contour_i < polyline.contours.size();
+         contour_i++) {
+      auto contour = polyline.contours[contour_i];
+      size_t contour_start_point_i, contour_end_point_i;
+      std::tie(contour_start_point_i, contour_end_point_i) =
+          polyline.GetContourPointBounds(contour_i);
+
+      switch (contour_end_point_i - contour_start_point_i) {
```

Comment:
nit: I'd recommend changing this to an `if` statement since it's tricky having `continue` and `break` statements at the same level but having different things they are affecting because of the switch.

Reply:
wait ... what are the break/continue even doing? the break is breaking the switch, and the continue is ... continuing the loop 😕

Reply:
Yep, see! confusing haha.

Reply:
I think it was common lisp where you had to specify the form you were interrupting, so you'd have to say something like `(break for)` in order to break out of a for loop.

Reply:
Okay fixed